### DISCUSSION TITLE
When right-clicking to create an Apex Class, the new class should automatically go into the selected folder

### DISCUSSION THREAD

**Discussion by randi274** [forcedotcom/salesforcedx-vscode#5825] [Ideas]
**Is your feature request related to a problem? Please describe.**
We recently fixed #3572 to allow creating an apex class in a subdirectory, but our users are still prompted to provide a directory when using the command! The prompt menu doesn't make much sense under the nested structure. 

**Describe the solution you'd like**

- If I right-click the `classes` directory to create an Apex class, it shouldn't prompt for the directory, and should automatically create the class there.
- If I right-click a subdirectory of `classes`, it shouldn't prompt for directory either, and should create the class in the correct subdirectory.
- It should work on all types of machines, as always 😄 
- If I use the Command Palette to create an Apex class, it should prompt for the directory. 

**Describe alternatives you've considered**
It works now, it's just a little clunky since I've already selected a directory with a right-click action.

**Comment by kaustubhdapurkar** [comment]
Hi @randi274 , could you please take a look at this draft.
I already see multiple issues with the new code, but created this PR to get started on the enhancement -

1. I have created an new class that implements ParametersGatherer which could be avoided. I did this because changing the existing `SelectOutputDir` class had bigger impact.
2. I have removed the `initialized `variable, will need to incorporate it with my changes, but I don't know how
3. I am not really sure about the unused parameter `uris`, but this doesn't seem to work if I don't pass it. I couldn't find any supporting documentation on this.
4. Since the command only shows up when selecting the singular classes folder, I have not written this code in a way that handles when `sourceUri` is an array. This could be problematic.

There could be more issues that I am not seeing. Not very familiar with TS and vscode extension API.
Thanks in advance for your time 😊.
